game_engine.properties

```diff
@@ -3,4 +3,4 @@
 ## If updating this file, remember to save a backup
 ## first.
 
-engine_version = 1.9.0.0
+engine_version = 1.9.0
```

Comment:
If we do not modify anything, then`m_micro` would never get set, right?
Does it make sense to add ".dev" here and then in the build script replace that with the build number?

Reply:
Correct, m_micro is never set to anything.

I guess it could be removed entirely.

Reply:
When a build is created, it is set to the build number because game_engine.properties has that appended to it.

Reply:
Indeed, when would we ever have the ".dev" extension added? It seems like dev's would manually add that. I think it would be much more ideal for the code to handle that or for that to not even be a concern in the first place.

Reply:
@DanVanAtta Please see #2087 for background on why `.dev` was added.  As pointed out there, `.dev` is purely eye candy to scream to the end user "this is a private build," and can be removed if it does not offer any value.

FWIW, just so it's clear to everyone (because it wasn't to me until I implemented #2087), the build version generation is a bit convoluted.  On Travis, the process is

1. Read the `engine_version` property from `game_engine.properties`.
1. Append Travis build number as the version "qualifier" (currently the fifth position).
1. Pass the qualified version to Gradle in the `engineVersion` project property.
1. Gradle replaces the value of the `engine_version` property in `game_engine.properties` with the qualified version and writes the contents to `build/rootFiles/game_engine.properties`.

When a dev runs the Gradle build locally, steps (1), (2), and (3) do not occur because they are done as part of `.travis.yml` (and there's no concept of a build number).  In this case, Gradle simply defaults to use the contents of `game_engine.properties` and adds the `.dev` qualifier to note that the build came from something other than Travis.  Adding the `.dev` qualifier was an attempt to keep the Travis and non-Travis build version strings symmetric by always ensuring there is a fifth qualifier field.

I think the version qualifier would be more useful if it contained a bit more information to uniquely identify the build.  For example, including the commit hash and timestamp of the build: `1.9.0.0.dev-d3eacdf-20170927T214659`.  But, as I said in the PR, having such a detailed build qualifier would only be useful if we were regularly providing private builds to users for testing.  We probably do that less than 10 times a year.